### DISCUSSION TITLE
How do I export an optional dependency group?

### DISCUSSION THREAD

**Discussion by redjax** [pdm-project/pdm#2525] [Q&A] [answered]
I have a `pyproject.toml` file with an optional dependency group called "ci". This is the `pyproject.toml` file:

```
[project]
name = "python-starter"
version = "0.1.0"
description = "Default template for PDM package"
authors = [
    {name = "redjax", email = "none@none.com"},
]
dependencies = []
requires-python = "==3.11.*"
readme = "README.md"
license = {text = "MIT"}


[project.optional-dependencies]
ci = [
    "nox>=2023.4.22",
    "tox>=4.11.4",
]
[tool.pdm]
package-type = "application"

[tool.pdm.dev-dependencies]
dev = [
    "black>=23.12.1",
    "ruff>=0.1.9",
    "pytest>=7.4.3",
    "pytest-xdist>=3.5.0",
    "nox>=2023.4.22",
    "tox>=4.11.4",
]

```

The PDM command I'm using to export these requirements to a file called `requirements.ci.txt` is: `pdm export --group ci -o requirements.ci.txt --without-hashes`.

I expect the contents of `requirements.txt` to only have `nox` and `tox` (I realize this is redundant, I'm just testing both with a single repository) and whatever dependencies those tools have, but instead it seems to be exporting all of my requirements, *including* the optional dependencies, rather than *only* the optional dependencies. This is the `requirements.ci.txt` file I'm getting:

```
# This file is @generated by PDM.
# Please do not edit it manually.

argcomplete==3.2.1
black==23.12.1
cachetools==5.3.2
chardet==5.2.0
click==8.1.7
colorama==0.4.6
colorlog==6.8.0
distlib==0.3.8
execnet==2.0.2
filelock==3.13.1
iniconfig==2.0.0
mypy-extensions==1.0.0
nox==2023.4.22
packaging==23.2
pathspec==0.12.1
platformdirs==4.1.0
pluggy==1.3.0
pyproject-api==1.6.1
pytest==7.4.3
pytest-xdist==3.5.0
ruff==0.1.9
tox==4.11.4
virtualenv==20.25.0

```

Thank you in advance to anyone who might know what I'm doing wrong! :)

**Comment by frostming** [comment, marked as answer]
--prod --no-default
